test/k8s/sctp.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package k8sTest
+
+import (
+	"path/filepath"
+
+	. "github.com/onsi/gomega"
+
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+)
+
+var _ = Describe("K8sSCTPTest", func() {
+
+	var (
+		kubectl             *helpers.Kubectl
+		ciliumFilename      string
+		iperfYAML           string
+		policyAllowSCTPYAML string
+		policyBlockSCTPYAML string
+	)
+
+	BeforeAll(func() {
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		SCTPManifestDir := helpers.ManifestGet(kubectl.BasePath(), "sctp")
+		iperfYAML = filepath.Join(SCTPManifestDir, "iperf3.yaml")
+		policyAllowSCTPYAML = filepath.Join(SCTPManifestDir, "policy-allow-sctp.yaml")
+		policyBlockSCTPYAML = filepath.Join(SCTPManifestDir, "policy-no-sctp.yaml")
+
+		ciliumFilename = helpers.TimestampFilename("cilium.yaml")
+		DeployCiliumOptionsAndDNS(kubectl, ciliumFilename,
+			map[string]string{"sctp.enabled": "true"})
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport("cilium endpoint list", "cilium service list")
+		res := kubectl.Exec("kubectl describe nodes")
+		GinkgoPrint(res.CombineOutput().String())
+		res = kubectl.Exec("kubectl describe pods")
+		GinkgoPrint(res.CombineOutput().String())
+
+	})
+
+	JustAfterEach(func() {
+		kubectl.ValidateNoErrorsInLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	AfterEach(func() {
+		By("Deleting all resources created during test")
+		kubectl.Delete(iperfYAML)
+		// Both have the same resource name.
+		kubectl.Delete(policyBlockSCTPYAML)
+
+		By("Waiting for all pods to finish terminating")
+		ExpectAllPodsTerminated(kubectl)
+	})
+
+	AfterAll(func() {
+		UninstallCiliumFromManifest(kubectl, ciliumFilename)
+		kubectl.CloseSSHClient()
+	})
+
+	It("Tests SCTP", func() {
```

Comment:
> So my awesome SCTP test depends on `CONFIG_IP_SCTP`, which COS does not enable :) What is the preferred approach for selectively disabling this test?

This should skip the test on GKE:


```suggestion
	SkipItIf(helpers.RunsOnGKE, "Tests SCTP", func() {
```

Reply:
Thanks!